src/copilot-cli/skills/session-end/scripts/rework_warning.py

```
@@ -0,0 +1,140 @@
+
+    Degrades gracefully:
+      - Git not available -> returns ``[]``
+      - Branch base not reachable (e.g. fresh clone) -> returns ``[]``
+      - Empty output (no commits ahead of base) -> returns ``[]``
+
+    Stricter/looser/different than canonical:
+      - This is a NEW check; there is no upstream validator to mirror.
+      - Threshold-6 is local-only; kill-criteria pattern documented inline.
+    """
+    try:
+        result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)